connector_clickup/models/clickup_backend/common.py

```diff
@@ -0,0 +1,364 @@
+        for backend in self.sudo():
+            backend._export_from_date(
+                model="clickup.project.tasks",
+                from_date_field=None if not from_sync else False,
+                with_delay=with_delay,
+            )
+
+    def _export_from_date(
+        self,
+        model,
+        from_date_field=None,
+        filters=None,
+        force_update_field=None,
+        priority=None,
+        with_delay=True,
+    ):
+        """Common method for export data from from_date."""
+
+        if not filters:
+            filters = {}
+        import_start_time = datetime.now()
+        job_options = {}
+        if priority or priority == 0:
+            job_options["priority"] = priority
+        for backend in self:
+            from_date = None
+            if from_date_field:
+                from_date = backend[from_date_field]
+                if from_date:
+                    from_date = from_date.strftime("%Y-%m-%dT%H:%M:%SZ")
+                else:
+                    from_date = to_iso_datetime(from_date)
+
+            filters["with_count"] = "true"
+            force = False
+            if force_update_field:
+                force = backend[force_update_field]
+
+            model_parts = model.split(".")
+            model_name = " ".join(part.title() for part in model_parts[1:])
+            model_name = " ".join(dict.fromkeys(model_name.split()))
+            job_options["description"] = f"Export Batch of Clickup {model_name}"
+
+            clickup_model = (
+                self.env[model].with_delay(**job_options or {})
+                if with_delay
+                else self.env[model]
+            )
+
+            clickup_model.export_batch(
+                backend,
+                filters=filters,
+            )
+            if force:
+                backend[force_update_field] = False
+        next_time = import_start_time - timedelta(seconds=IMPORT_DELTA_BUFFER)
+        if from_date_field:
+            self.write({from_date_field: next_time})
+
+    def cron_import_clickup_changes(
+        self,
+        filters=None,
+        with_delay=True,
+    ):
+        """Scheduled action method of import"""
+        data = self.env["clickup.backend"].search([])
+
+        for backend in data:
+            clickup_model = backend.env["clickup.project.project"].with_delay()
+            clickup_model.import_batch(
+                backend,
+                filters=filters,
+                force=True,
+                **{"no_delay": not with_delay},
+            )
+        for backend in data:
+            clickup_model = backend.env["clickup.project.task.type"].with_delay()
+            clickup_model.import_batch(
+                backend,
+                filters=filters,
+                force=True,
+                **{"no_delay": not with_delay},
+            )
+        for backend in data:
+            clickup_model = backend.env["clickup.project.tasks"].with_delay()
```

Comment:
I think when we use with_delay directly, the priority will not be set in the task (for batch) in queue jobs.